docs/branches.md

```diff
@@ -52,6 +52,26 @@ some remote.
 If you want to know the internals of branch tracking, consult the 
 [Design Doc][design].
 
+### Terminology summary
+
+- A **remote branch** is a branch ref on the remote. `jj` can find out its state
+  only when it's actively communicating with the remote. 
+- A **tracked (remote) branch** is defined above. You can make a remote branch
+  tracked with the [`jj branch track` command](#manually-tracking-a-branch), for
+  example.
+- A **local tracking branch** is the local branch that `jj` tries to keep in
+  sync with the tracked remote branch. For example, after `jj branch track
+  develop@origin`, there will be a local branch `develop` that's tracking the
+  remote `develop@origin` branch.
+- A **remote-tracking branch** is a term common in the Git world, but `jj` has
+  them too. The term corresponds to the label of the last-remembered position of
+  a remote branch. For example, this is the commit that `jj checkout
```

Comment:
For me it was hard to parse the second sentence, as it is not clear about which VCS we're talking about. Can you make it a bit more explicit that the term description is for `jj`. 

While I was writing the above, I noticed this: isn't the `remote-tracking branch` just a bookmark to the last known remote commit? So wouldn't the term `remote-tracking branch-ref` be more accurate?

Reply:
> For me it was hard to parse the second sentence, as it is not clear about which VCS we're talking about. Can you make it a bit more explicit that the term description is for jj.

I can't think of an improvement right now, will look at it later. Perhaps I should be less aggressive about labeling this as a "Git term".

> While I was writing the above, I noticed this: isn't the remote-tracking branch just a bookmark to the last known remote commit? So wouldn't the term remote-tracking branch-ref be more accurate?

In my mind, the answer to your first question is "yes". I think of "branches" as a kind of "ref", and thus equivalent to "branch-refs". So, the term "remote-tracking branch-ref" sounds accurate to me, but completely equivalent, and I'm not sure it's clearer.

It's good you pointed this out; it does feel like a weird point of terminology that can confuse people with a different mindset from mine. Again, right now, I'm not sure how to improve it.

Reply:
> I can't think of an improvement right now, will look at it later. Perhaps I should be less aggressive about labeling this as a "Git term".

Yes, that would be a improvement. Yuya has a good point in https://github.com/martinvonz/jj/pull/2772#discussion_r1442427271 too. 

> In my mind, the answer to your first question is "yes". I think of "branches" as a kind of "ref", and thus equivalent to "branch-refs". So, the term "remote-tracking branch-ref" sounds accurate to me, but completely equivalent, and I'm not sure it's clearer.

Currently branches are just "refs" for `jj`, as we don't have something which describes a named set of commits (like topics could be), and is the reason I used bookmark at first.  

I'd be in favor of calling it just the "last known remote revision" and then exposing it as such in the commit templater. 

> It's good you pointed this out; it does feel like a weird point of terminology that can confuse people with a different mindset from mine. Again, right now, I'm not sure how to improve it.

I don't think we have a huge missmatch in our concepts but as most users come from Git, for them the current phrasing is highly confusing. And since `jj` has nothing Git branch-like, this will only lead to more questions.

As @necauqua eloquently put it in Discord: 
> git gets away with it because it doesn't show non-bookmarked commits, so topological branch and the bookmark thing are kind of the same, which is slightly confusing on it's own btw when teaching people about stuff like hard resets

Git currently just get's away with.

Reply:
I did deemphasize that it's a "Git term"

For the rest of this discussion, I think this is a little outside the scope of this PR. At least, I don't have an idea of how to improve it within the scope of the PR. There's an argument to be made that our "named branches" should not be called branches at all. I'm not sure that "ref" is a perfect name, since ideally I'd want to distinguish between refs that move with rebased commits and refs that don't, but there's an argument to be made that it's better than "branch".

However, in my mind, as long as we use the name "branches" following Git, mentioning "remote-tracking branches" is OK.

Reply:
> I did deemphasize that it's a "Git term"

Thanks. 

> For the rest of this discussion, I think this is a little outside the scope of this PR.

Agreed. 

> There's an argument to be made that our "named branches" should not be called branches at all. I'm not sure that "ref" is a perfect name, since ideally I'd want to distinguish between refs that move with rebased commits and refs that don't, but there's an argument to be made that it's better than "branch".

I honestly think our "named branches" should just be called bookmarks, as martin proposed. The term "ref" mixes a technical (implementation) term with a user-visible term, which is honestly quite bad. All in all I think `jj` does not have something like Git's branches. 

> However, in my mind, as long as we use the name "branches" following Git, mentioning "remote-tracking branches" is OK.

I agree. I think we can put a lid on the "branch" naming conversation here and move it either to Discord or a issue.